src/txreconciliation.h

```diff
@@ -0,0 +1,585 @@
+
+/** The size of the field, used to compute sketches to reconcile transactions (see BIP-330). */
+static constexpr unsigned int RECON_FIELD_SIZE = 32;
+/**
+ * Allows to infer capacity of a reconciliation sketch based on it's char[] representation,
+ * which is necessary to deserealize a received sketch.
+ */
+static constexpr unsigned int BYTES_PER_SKETCH_CAPACITY = RECON_FIELD_SIZE / 8;
+/** Limit sketch capacity to avoid DoS. */
+static constexpr uint16_t MAX_SKETCH_CAPACITY = 2 << 12;
+/**
+* It is possible that if sketch encodes more elements than the capacity, or
+* if it is constructed of random bytes, sketch decoding may "succeed",
+* but the result will be nonsense (false-positive decoding).
+* Given this coef, a false positive probability will be of 1 in 2**coef.
+*/
+static constexpr unsigned int RECON_FALSE_POSITIVE_COEF = 16;
+static_assert(RECON_FALSE_POSITIVE_COEF <= 256,
+    "Reducing reconciliation false positives beyond 1 in 2**256 is not supported");
+/** Default coefficient used to estimate set difference for tx reconciliation. */
+static constexpr double DEFAULT_RECON_Q = 0.02;
+/** Used to convert a floating point reconciliation coefficient q to an int for transmission.
+  * Specified by BIP-330.
+  */
+static constexpr uint16_t Q_PRECISION{(2 << 14) - 1};
+/**
+ * Interval between sending reconciliation request to the same peer.
+ * This value allows to reconcile ~100 transactions (7 tx/s * 16s) during normal system operation
+ * at capacity. More frequent reconciliations would cause significant constant bandwidth overhead
+ * due to reconciliation metadata (sketch sizes etc.), which would nullify the efficiency.
+ * Less frequent reconciliations would introduce high transaction relay latency.
+ */
+static constexpr std::chrono::microseconds RECON_REQUEST_INTERVAL{16s};
+/**
+ * Interval between responding to peers' reconciliation requests.
+ * We don't respond to reconciliation requests right away because that would enable monitoring
+ * when we receive transactions (privacy leak).
+ */
+static constexpr std::chrono::microseconds RECON_RESPONSE_INTERVAL{2s};
+
+/**
+ * Represents phase of the current reconciliation round with a peer.
+ */
+enum ReconciliationPhase {
+    RECON_NONE,
+    RECON_INIT_REQUESTED,
+    RECON_INIT_RESPONDED,
+    RECON_EXT_REQUESTED,
+    RECON_EXT_RESPONDED
+};
+
+/**
+ * This struct is used to keep track of the reconciliations with a given peer,
+ * and also short transaction IDs for the next reconciliation round.
+ * Transaction reconciliation means an efficient synchronization of the known
+ * transactions between a pair of peers.
+ * One reconciliation round consists of a sequence of messages. The sequence is
+ * asymmetrical, there is always a requestor and a responder. At the end of the
+ * sequence, nodes are supposed to exchange transactions, so that both of them
+ * have all relevant transactions. For more protocol details, refer to BIP-0330.
+ */
+class ReconciliationState {
```

Comment:
What do you think about using a pimpl model (like txrequest)? That way, the data members of `TxReconciliationTracker` could be placed in the implementation file, including `m_states`. `ReconciliationState` could be moved entirely into the cpp file.